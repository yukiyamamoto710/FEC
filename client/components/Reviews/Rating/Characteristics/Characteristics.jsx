import React from 'react';
import PropTypes from 'prop-types';

const Characteristics = (props) => {
  const { value, dataArray, name } = props;
  const perce = Math.floor((value / 5) * 100);
  const point = {
    position: 'absolute',
    bottom: '10%',
    left: `${perce}%`,
  };

  return (
    <div
      data-testid="chara"
      className="charaContainer"
    >
      {name}
      <div className="charaBar">
        <div className="charaSmallBar" />
        <div className="charaSmallBar" />
        <div className="charaSmallBar" />
        <div className="charaSmallBar" />
        <div className="charaSmallBar" />
        <div style={point}>
          <div className="charaBarPoint" />
        </div>
      </div>
      <div className="barTextContainer">
        <div className="barText">{dataArray[0]}</div>
        <div className="barText">{dataArray[dataArray.length - 1]}</div>
      </div>
    </div>
  );
};

Characteristics.propTypes = {
  name: PropTypes.string,
  dataArray: PropTypes.arrayOf(PropTypes.string),
  value: PropTypes.number,
};

Characteristics.defaultProps = {
  name: '',
  dataArray: [],
  value: 0,
};

export default Characteristics;
