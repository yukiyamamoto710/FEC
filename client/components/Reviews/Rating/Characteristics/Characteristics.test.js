/**
 * @jest-environment jsdom
 */
import React from 'react';
import renderer from 'react-test-renderer';
import {
  render,
  screen,
  cleanup,
} from '@testing-library/react';
import '@testing-library/jest-dom';
import Characteristics from './Characteristics';

afterEach(cleanup);

it('should render', () => {
  render(<Characteristics name="hello World" />);
  const test = screen.queryByText('hello World');
  expect(test).toBeInTheDocument();
});

it('should render', () => {
  const array = ['hi', 'Hello', 'World'];
  render(<Characteristics dataArray={array} />);
  const test = screen.queryByText('hi');
  const test1 = screen.queryByText('Hello');
  const test2 = screen.queryByText('World');
  expect(test).toBeInTheDocument();
  expect(test1).not.toBeInTheDocument();
  expect(test2).toBeInTheDocument();
});

it('matches snapShot', () => {
  const tree = renderer.create(
    <Characteristics
      name="hi"
      dataArray={['hi', 'Hello', 'World']}
      value={10}
    />,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
