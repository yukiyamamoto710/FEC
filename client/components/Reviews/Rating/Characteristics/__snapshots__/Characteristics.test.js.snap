// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapShot 1`] = `
<div
  className="charaContainer"
  data-testid="chara"
>
  hi
  <div
    className="charaBar"
  >
    <div
      className="charaSmallBar"
    />
    <div
      className="charaSmallBar"
    />
    <div
      className="charaSmallBar"
    />
    <div
      className="charaSmallBar"
    />
    <div
      className="charaSmallBar"
    />
    <div
      style={
        Object {
          "bottom": "10%",
          "left": "200%",
          "position": "absolute",
        }
      }
    >
      <div
        className="charaBarPoint"
      />
    </div>
  </div>
  <div
    className="barTextContainer"
  >
    <div
      className="barText"
    >
      hi
    </div>
    <div
      className="barText"
    >
      World
    </div>
  </div>
</div>
`;
